Backend/app/controllers/events_controller.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show update destroy]
+  before_action :set_user, only: %i[index create]
+  before_action :set_room_booking, only: %i[create]
+ 
+  # GET /events
+  def index
+    @events = Event.all
+    unless params[:event_type].blank? 
+    @events=@events.where(:event_type =>params[:event_type])
+    end
+    render json: @events
+
+  end
+
+  # GET /events/1
+  def show
+    render json: @event
+  end
+
+  # POST /events
+  def create
+    if @room_booking.event.blank?
+    @event = Event.new(event_params)
```

Comment:
I don't think you need `@room_booking`, you can just also permit room_booking_id in the event params and have everything there. And I may have missed something. What is event_type? If it's always gonna be the same as room_type, we don't need it.

Reply:
First of all, thank you for your time.
I thought event type means if this event gonna be a conference or lecture? Am I understood wrong?

Reply:
no problem, that's what I'm here for :)
Makes sense, but isn't room type class/lecture hall, and event type lecture/conference which could be in any type of room?

Reply:
I think they're the same room_type and event_type, and that's why I build that way, but if event_type must be changed from UI- so what will be written there?